lib/jmap/mail/email/submission/email_submission.dart

```diff
@@ -1,21 +1,24 @@
 import 'package:equatable/equatable.dart';
 import 'package:jmap_dart_client/http/converter/delivery_status_converter.dart';
-import 'package:jmap_dart_client/http/converter/email_id_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/email_submission_id_nullable_converter.dart';
 import 'package:jmap_dart_client/http/converter/id_converter.dart';
 import 'package:jmap_dart_client/http/converter/id_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/reference_email_id_nullable_converter.dart';
 import 'package:jmap_dart_client/http/converter/thread_id_nullable_converter.dart';
 import 'package:jmap_dart_client/http/converter/undo_status_nullable_converter.dart';
 import 'package:jmap_dart_client/http/converter/utc_date_nullable_converter.dart';
 import 'package:jmap_dart_client/jmap/core/id.dart';
 import 'package:jmap_dart_client/jmap/core/utc_date.dart';
 import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:jmap_dart_client/jmap/mail/email/reference_email_id.dart';
 import 'package:jmap_dart_client/jmap/mail/email/submission/delivery_status.dart';
+import 'package:jmap_dart_client/jmap/mail/email/submission/email_submission_id.dart';
 import 'package:jmap_dart_client/jmap/mail/email/submission/envelope.dart';
 
 class EmailSubmission with EquatableMixin {
-  final Id? id;
+  final EmailSubmissionId? id;
   final Id? identityId;
-  final EmailId? emailId;
+  final ReferenceEmailId? emailId;
```

Comment:
IMO, `EmailId` is ok
```
class EmailId(Id)
```

Reply:
It should be a reference.

Reply:
IMO, `EmailId` is enough, because `EmailId(Id)` and `ReferenceId` extend from `Id`